dttools/src/list.c

```diff
@@ -1,406 +1,633 @@
 /*
-Copyright (C) 2003-2004 Douglas Thain and the University of Wisconsin
-Copyright (C) 2005- The University of Notre Dame
+Copyright (C) 2018- The University of Notre Dame
 This software is distributed under the GNU General Public License.
 See the file COPYING for details.
 */
 
+#include <assert.h>
+#include <limits.h>
+#include <math.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
 #include "list.h"
 
-#include <stdlib.h>
+struct list {
+	unsigned refcount;
+	unsigned length;
+	struct list_item *head;
+	struct list_item *tail;
+	struct list_cursor *iter;
```

Comment:
Add a comment that this the global iterator, used for backwards compatibility, etc.